### PR TITLE
User dropdown

### PR DESCRIPTION
**Pull request by Schottkyc137**
see issue: #681 

### Visible/Frontend Changes
- [x] When selecting a user, a list of possibilities with all users will show up
- [x] Multiple authors can be chosen when looking at a note 

### Architectural/Backend Changes
- [x] The note-details component uses the new field compared to the old one
- [x] Notes has an array of authors (strings)

### CODE REVIEW

File: src/app/core/user/user-select/user-select.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { AlertService } from "../../alerts/alert.service";
+import { LoggingService } from "../../logging/logging.service";
+
+@Component({
+  selector: "app-user-select",
+  templateUrl: "./user-select.component.html",
+  styleUrls: ["./user-select.component.scss"],
+})
+export class UserSelectComponent implements OnInit {
+  @Input() user: string;
```
Comment (sleidig) [inline src/app/core/user/user-select/user-select.component.ts:21]: To make this more generally powerful we should probably use `User` entity objects (or maybe id) in the input and output. Then the code using this component can decide which fields are needed at a higher level.

We'll probably need to do a simple migration of old data, mapping names to ids and save user ids with the note, but that should not be a big issue.

File: src/app/core/user/user-select/user-select.component.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  @Output() userChange = new EventEmitter<string>();
```
Comment (sleidig) [inline src/app/core/user/user-select/user-select.component.ts:22]: That was not clearly specified in the issue, but we have cases where more than one user is selected so we will need to support multi-select somehow

File: src/app/core/user/user-select/user-select.component.ts (hunk 1 above)
Comment (Schottkyc137) [reply src/app/core/user/user-select/user-select.component.ts:21]: So to just to clarify, a Note should therefore not only have one author (which currently is a string) but multiple authors (which should be an array of users, resp. an array of user IDs)?

File: src/app/core/user/user-select/user-select.component.ts (hunk 2 above)
Comment (sleidig) [reply src/app/core/user/user-select/user-select.component.ts:22]: The Note "author" is actually misnamed currently, it is rather used as a description of staff members (= users) who have been part of the real-world interaction the Note describes. Therefore more than one user can be linked.

File: src/app/core/user/user-select/user-select.component.ts (hunk 1 above)
Comment (sleidig) [reply src/app/core/user/user-select/user-select.component.ts:21]: Yes, that should be the it.

And maybe we can abstract this (now or later) into a generic base component also, so the UX and implementation is shared between child and user select field (#136)

**Comment by Schottkyc137** [comment]
Closed since this is in PR #728

**Pull request closed without merging**